src/components/metadataSelects/CategoryComboSelect.tsx

```diff
@@ -0,0 +1,132 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { SearchableSingleSelect } from '../SearchableSingleSelect'
+
+interface CategoryComboQueryResult {
+    categoryCombos: {
+        pager: {
+            pageCount: number,
+            pageSize: number,
+            page: number,
+            total: number,
+        },
+        categoryCombos: Array<{
+            id: string,
+            displayName: string,
+        }>,
+    }
+}
+
+type QueryParams = {
+    page: number,
+    pageSize: number,
+    fields: string[],
+    filter?: string,
+}
+
+const query = {
+    categoryCombos: {
+        resource: 'categoryCombos',
+        params: ({ ...dynamicParams }) => {
+            const baseParams: QueryParams = {
+                page: dynamicParams.page,
+                pageSize: 20,
+                fields: ['id', 'displayName'],
+            }
+
+            if (dynamicParams.filter) {
+                baseParams.filter = `displayName:ilike:${dynamicParams.filter}`
+            }
+
+            return baseParams
+        },
+    },
+}
+
+interface Option { value: string, label: string }
+export function CategoryComboSelect({
+    onChange,
+    selected,
+}: {
+    onChange: ({ value, label }: Option) => void,
+    // Must be an option as we always must supply the selected option, but the
+    // list might not contain it when filtering / loading the first page of
+    // options
+    selected?: Option,
+}) {
+    // Using a ref because we don't want to react to a change of this value
+    // It's guaranteed that we have the correct value because we reset the
+    // value before fetching a new set of options when filtering / resetting
+    // the filter as well as setting the value after the fetch has completed
+    // but before we call a state setter. The state setter will cause a
+    // rerender during which the new pages value can be accessed already
+    const pages = useRef(1)
+    const [loadedOptions, setLoadedOptions] = useState<Array<{
+      value: string,
+      label: string,
+    }>>([])
+
+    const [params, setParams] = useState({ page: 1, filter: '' })
```

Comment:
I don't think we need a state for this? 
I think `adjustQueryParamsWithChangedFilter` should just call refetch, with `{ params: { page: 1, filter: value}`.

`IncrementPage` will do the same. This should hopefully fix the flashing list. And we don't need the use-effect.

Reply:
I removed the state :ok_hand: